src/datasources/entity-ds/datasource.js

```diff
@@ -353,19 +353,19 @@ export class OpenNMSEntityDatasource {
     }
 
     acknowledgeAlarm(alarmId) {
-        return this.opennmsClient.doAck(alarmId, this.user);
+        return this.q.when(this.opennmsClient.doAck(alarmId, this.user));
     }
 
     unacknowledgeAlarm(alarmId) {
-        return this.opennmsClient.doUnack(alarmId, this.user);
+        return this.q.when(this.opennmsClient.doUnack(alarmId, this.user));
     }
 
     clearAlarm(alarmId) {
-        return this.opennmsClient.doClear(alarmId, this.user);
+        return this.q.when(this.opennmsClient.doClear(alarmId, this.user))
```

Comment:
Can't we use the existing promised returned by `doClear` instead of creating a new one?

Reply:
Yes, we should use the same.. this is extra now. Updated the PR in https://github.com/OpenNMS/opennms-helm/pull/254/commits/2f63d6b3d3b28c3a09a77158f3843b0d246f6138
Thanks! :)